networkit/cpp/graph/Graph.cpp

```diff
@@ -1065,4 +1068,52 @@ Graph Graph::subgraphFromNodes(const std::unordered_set<node> &nodes) const {
 	return S;
 }
 
+void Graph::toLargestConnectedComponent() {
+	if (this->isDirected())
+		throw std::runtime_error(
+			"Method available for undirected graphs only.");
+
+	ConnectedComponents cc(*this);
+	cc.run();
+	auto comps = cc.getComponents();
+	std::sort(comps.begin(), comps.end(),
+		[&](const std::vector<node> &x, const std::vector<node> &y) {
+			return x.size() > y.size();
+	});
+	std::unordered_map<node, node> map;
+	std::unordered_set<node> set(comps[0].begin(), comps[0].end());
+	node idx = 0;
+	for (node u : comps[0])
+		map[u] = idx++;
+	std::vector<std::pair<node, node>> edges;
+	std::vector<edgeweight> weights;
+	this->subgraphFromNodes(set).forEdges([&](node x, node y, edgeweight w) {
+		if (map[x] != map[y]) {
+			edges.push_back(std::make_pair(map[x], map[y]));
+			if (this->isWeighted())
+				weights.push_back(w);
+		}
+	});
+
+	n = idx;
+	z = n;
+	m = 0;
+	storedNumberOfSelfLoops = 0;
+	outEdges.clear();
+	outEdges.resize(n);
+	inEdges.clear();
+	inEdges.resize(n);
+	inEdgeWeights.clear();
+	inEdgeWeights.resize(n);
+	outEdgeWeights.clear();
+	outEdgeWeights.resize(n);
+	exists.clear();
+	exists.resize(n, true);
+
+	for (count e = 0; e < edges.size(); ++e) {
+		this->addEdge(edges[e].first, edges[e].second);
+		if (this->isWeighted())
+			this->setWeight(edges[e].first, edges[e].second, weights[e]);
```

Comment:
This innocent-looking `setWeight` is an O(deg(u)) complexity! Why does the code not simply give the weight to `addEdge`?